zkevm-circuits/src/evm_circuit/execution/memory_copy.rs

```diff
@@ -0,0 +1,532 @@
+use crate::{
+    evm_circuit::{
+        execution::ExecutionGadget,
+        param::MAX_MEMORY_SIZE_IN_BYTES,
+        step::ExecutionState,
+        table::TxContextFieldTag,
+        util::{
+            constraint_builder::{
+                ConstraintBuilder, StepStateTransition, Transition::Delta,
+            },
+            math_gadget::ComparisonGadget,
+            memory_gadget::BufferGetDataGadget,
+            Cell,
+        },
+        witness::{Block, Call, ExecStep, OpcodeExtraData, Transaction},
+    },
+    util::Expr,
+};
+use halo2::{arithmetic::FieldExt, circuit::Region, plonk::Error};
+
+const MAX_COPY_BYTES: usize = 39;
```

Comment:
Could you mind adding comment here to describe how the `MAX_COPY_BYTES` come out for better readability ?

Reply:
Added comment